### DISCUSSION TITLE
@Key Fields

### DISCUSSION THREAD

**Discussion by runningjustin** [babyfish-ct/jimmer#611] [Q&A]
What is the right way to update 1+ rows that all share a same value. e.g. `update user set status='VERIFIED' where kycStatus='VERIFIED'`. If I use update, I'd have something like this:

```
sqlClient.update(new(User::class).by {
    this.kycStatus = "VERIFIED"
    this.status = "VERIFIED"
})
```

I want to use update() instead of createUpdate() because I have beforeSave() interceptors I need to run. If I flag kycStatus as a `@Key` field, I get the error `cannot guarantee uniqueness under that path, do you forget to add unique constraint for that key?` This leads me to believe that `@Key` fields must be unique, but in this case it is not.

How can I achieve this update? I did just read about `setKeyProps()` but I'm not sure if that would work (and doesn't seem to be in the version I'm using so I can't test it).

Thanks...

**Comment by babyfish-ct** [comment]
There is a member function `setOptimisticLock` in the save comand configuration, but I forget to wrap it in kotlin API so that you cannot see it, please wait for next version.

**Comment by babyfish-ct** [comment]
Try `0.8.150`

Save function has a function `setOptmitiscLock`
